### ISSUE TITLE
`onAfterRenderPage` fires only once

### ISSUE THREAD

**Issue by akhoury** [surveyjs/surveyjs#664]
### Are you requesting a feature, reporting a bug or ask a question?
It sounds like a bug, the [docs](https://surveyjs.io/Documentation/Library/?id=surveymodel#onAfterRenderPage) says 

> The event is fired right after a page is rendred in DOM. Use it to modify html elements. There are two parameters in options: options.currentPage, options.htmlElement

it sounds like we can use to modify each page after it's rendered. It also provides a `currentPage`

### What is the current behavior?

`onAfterRenderPage` is firing only once on page0

### What is the expected behavior?

`onAfterRenderPage` should fire on every page render

### How would you reproduce the current behavior (if this is a bug)?

create a survey with multiple pages, add an `onAfterRenderPage` handler and try it.

#### Provide the test code and the tested page URL (if applicable)

Tested page URL: https://plnkr.co/edit/5a1w1O5tJhdfKIwslYCo

Test code

```javascript
Survey.Survey.cssType = "bootstrap";
Survey.defaultBootstrapCss.navigationButton = "btn btn-green";

window.survey = new Survey.Model({
 "pages": [
  {
   "name": "page1",
   "questions": [
    { "type": "text", "name": "satisfaction", "title": "How satisfied are you with the Product?" }
   ]
  },
  {
   "name": "page2",
   "questions": [
    { "type": "text", "name": "why", "title": "why?" }
   ]
  }
 ]
});
survey.onComplete.add(function(result) {
    document.querySelector('#surveyResult').innerHTML = "result: " + JSON.stringify(result.data);
});

survey.onAfterRenderPage .add(function(result, options) {
  alert(options.page.name);
});

survey.data = {satisfaction: 'ok, i guess', why: 'because'};

$("#surveyElement").Survey({ 
    model: survey 
});
```

### Specify your

* browser: `Chrome`
* browser version: `61.0.3163.91`
* surveyjs platform (angular or react or jquery or knockout or vue): `jquery`
* surveyjs version: `0.12.23`

**Comment by akhoury** [comment]
ok so `onCurrentPageChanged` works on every page change, however it doesn't fire for the initial page. 

# workaround

use both, `onAfterRenderPage` which fires for page0, and `onCurrentPageChanged` which fires for every page after that.

**Comment by andrewtelnov** [comment]
@akhoury Yes, right. If nothing changed in the page model, the afterRenderPage event is not fired. So, I would say it is not a workaround, I would recommend to use the same.

Thank you,
Andrew

**Comment by akhoury** [comment]
I am trying to execute some logic at every page. including the first one. 
(basically updating the url with `#currentPageNo=N`)  so if someone refreshes the page, they would land at the last one they saw and not have to start over.

**Comment by andrewtelnov** [comment]
@akhoury Do you have this url inside or outside the survey?

**Comment by akhoury** [comment]
outside. 
I am literally modifying the `location.href` using `history.pushState`

**Comment by andrewtelnov** [comment]
@akhoury In this case all you need is survey.currentPageNo property and onCurrentPageChanged event.
Initially survey.currentPageNo is 0, unless you do not change it in the code.

Thank you,
Andrew

**Comment by akhoury** [comment]
ok, that's fine. but is `onAfterRenderPage` working as designed or is it supposed to be fire at every page render?

**Comment by andrewtelnov** [comment]
@akhoury onAfterRenderPage fired when the page needs to be rendered or re-rendered. If page model doesn't changed, it shows again, used previous generated elements.

Thank you,
Andrew

**Comment by andrewtelnov** [comment]
@akhoury I have revisted this issue. It should be fine now.

Thank you,
Andrew

**Comment by yakou32** [comment]
Sorry to come back on this topic.
I was also expecting `onAfterRenderPage` to fire after each page render (or re-render), including the first page render... And it looks like first page rendering is an exception where `onAfterRenderPage` is not rendered.

**Comment by dmitrykurmanov** [comment]
@yakou32 do you need an another bahaviour? And if yes, please describe your use case.

**Comment by yakou32** [comment]
I finally solved my personal use case differently.

But anyway, this issue is about being able to customize the markup, including the first page markup.
Here is a potential use case: we would like to display a dynamic matrix using barrating for `select`.
If the user adds additional rows that include `select` dropdowns, they will not be rendered properly as jQuery barrating should be refired after DOM update.
This is the way it works for any page, but it won't work on the first page, as if it was impossible to use `onAfterRenderPage()`. Therefore, it looks inconsistent and developers who need it on first page may face the issue...

**Comment by nrhoffmann** [comment]
What's the deal with this issue?
I'm using React and found that `onCurrentPageChanged` fired before the html that represents the next page is rendered.

**Comment by dmitrykurmanov** [comment]
@nrhoffmann I've created a separate issue for your question: https://github.com/surveyjs/surveyjs/issues/1505

**Comment by the-glima** [comment]
Still `onAfterRenderPage` doesn't fired when first page is rendered, I didn't find any other event to use when Survey is loaded also.

Tried a bunch of events `onStarted`, `onAfterRenderPage`, `onAfterRenderPanel`, etc... nothing is being fired at first load.
